### PR TITLE
cmd: Add support for benchmarking partial evaluation queries

### PR DESCRIPTION
**Pull request by tsandall**
This change allows users to get an accurate estimate of partial eval
performance on their policies.

Signed-off-by: Torin Sandall <torinsandall@gmail.com>

### CODE REVIEW

**Review by ashutosh-narkar — COMMENTED** [review]
The changes lgtm ! We seem to be not hitting the test coverage target.

File: cmd/bench.go [hunk 1]
```diff
@@ -15,6 +14,7 @@ import (
 	"testing"
 
 	"github.com/olekukonko/tablewriter"
+	"github.com/pkg/errors"
```
Comment (srenatus) [inline cmd/bench.go:17]: Just a note, I think we can do without this package from Golang 1.13 onwards, see https://blog.golang.org/go1.13-errors. `fmt.Errorf("yaddda: %w", err)` with the `%w` directive does what `errors.Wrap(err, "yadda")` does, without losing the information about the wrapped error.

I suppose we could refactor this in one sweep throughout our code base, not blocking this PR.

File: cmd/bench.go [hunk 2]
```diff
@@ -99,7 +100,7 @@ The optional "gobench" output format conforms to the Go Benchmark Data Format.
 }
 
 type benchRunner interface {
-	run(ctx context.Context, ectx *evalContext, pq rego.PreparedEvalQuery, params benchmarkCommandParams) (testing.BenchmarkResult, error)
+	run(ctx context.Context, ectx *evalContext, params benchmarkCommandParams, f func(context.Context, ...rego.EvalOption) error) (testing.BenchmarkResult, error)
```
Comment (srenatus) [inline cmd/bench.go:103]: [nit] we could name the function type, like `type benchFunc func(context.Context, ...rego.EvalOption) error`. But it has little impact outside of the use in `bench_test.go` below.

**Review by srenatus — APPROVED** [review]
LGTM :)

**Review by srenatus — APPROVED** [review]

File: cmd/bench.go [hunk 3]
```diff
@@ -15,13 +14,11 @@ import (
...
 
 	"github.com/olekukonko/tablewriter"
-	"github.com/spf13/cobra"
-
 	"github.com/open-policy-agent/opa/internal/presentation"
-
 	"github.com/open-policy-agent/opa/metrics"
 	"github.com/open-policy-agent/opa/rego"
 	"github.com/open-policy-agent/opa/util"
+	"github.com/spf13/cobra"
```
Comment (srenatus) [inline cmd/bench.go:21]: Well I had decided to turn a blind eye, the linter never does that, fortunately 😄 https://github.com/open-policy-agent/opa/pull/3049/checks?check_run_id=1665309105#step:3:180

**Review by srenatus — APPROVED** [review]

**Pull request merged**